Modules/Forum/classes/class.ilForumTopic.php

```diff
@@ -565,40 +565,43 @@ public function movePosts($old_obj_id, $old_pk, $new_obj_id, $new_pk)
 				}
 			}
 
-			$this->db->lockTables(
-				array(
-					0 => array('name' => 'frm_user_read',     'type' => ilDBConstants::LOCK_WRITE),
-					1 => array('name' => 'frm_thread_access', 'type' => ilDBConstants::LOCK_WRITE)
-				)
-			);
+			$current_id = $this->id;
 
-			$this->db->manipulateF('
+			$ilAtomQuery = new ilAtomQuery($ilDB);
+			$ilAtomQuery->addTable('frm_user_read', ilAtomQuery::LOCK_WRITE);
+			$ilAtomQuery->addTable('frm_thread_access', ilAtomQuery::LOCK_WRITE);
+			$ilAtomQuery->addQueryClosure(function ($ilDB) use ($new_obj_id, $current_id) {
+				$ilDB->manipulateF('
 				DELETE FROM frm_user_read
 				WHERE obj_id = %s AND thread_id =%s',
-				array('integer', 'integer'),
-				array($new_obj_id, $this->id));
-			
-			$this->db->manipulateF('
+					array('integer', 'integer'),
+					array($new_obj_id, $current_id));
+			});
+			$ilAtomQuery->addQueryClosure(function ($ilDB) use ($new_obj_id, $current_id) {
```

Comment:
I splitted the queries into multiple Closures which should allow to rerun only to not failed queries (will follow in the next commit)